var/spack/repos/builtin/packages/singularityce/package.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+import llnl.util.tty as tty
+import os
+import shutil
+
+
+class SingularityBase(MakefilePackage):
+    '''Singularity is a container technology focused on building portable
+       encapsulated environments to support "Mobility of Compute" For older
+       versions of Singularity (pre 3.0) you should use singularity-legacy,
+       which has a different install base (Autotools).
+
+       Needs post-install chmod/chown steps to enable full functionality.
+       See package definition or `spack-build-out.txt` build log for details,
+       e.g.
+
+       tail -15 $(spack location -i singularity)/.spack/spack-build-out.txt
+    '''
+
+    version('3.7.4', sha256='c266369a8bf2747f44e0759858c3fc3b2325b975a8818b2668f0b97b124d0164')
+    version('3.7.3', sha256='6667eb8875d2b66d73504f40c956b42b1351744f488d164204376215d885da5c')
+    version('3.7.2', sha256='36916222e26fb934404f0766e0ff368edac36d7fc31ca571f5f609466609066b')
+    version('3.7.1', sha256='82d2c65063560195ec34551931be3c325b95e8e2009e92755fd7daad346e083c')
+    version('3.7.0', sha256='fb96aaf5f462a56a4a5bd2951287bcbbefe8cf543e228e4e955428f386a8d478')
+    version('3.6.4', sha256='71233a81d6bb4d686d8dc636b3e3e962a372f54001921c89a12b062cefd9e79f')
+    version('3.6.3', sha256='b1a985757a9907d8db0f102fc170a25387e715f7ff31957be964bf47914ea2fd')
+    version('3.6.2', sha256='dfd7ec7376ca0321c47787388fb3e781034edf99068f66efc36109e516024d9b')
+    version('3.6.1', sha256='6cac56106ee7f209150aaee9f8788d03b58796af1b767245d343f0b8a691121c')
+    version('3.5.3', sha256='0c76f1e3808bf4c10e92b17150314b2b816be79f8101be448a6e9d7a96c9e486')
+    version('3.5.2', sha256='f9c21e289377a4c40ed7a78a0c95e1ff416dec202ed49a6c616dd2c37700eab8')
+    version('3.4.1', sha256='638fd7cc5ab2a20e779b8768f73baf21909148339d6c4edf6ff61349c53a70c2')
+    version('3.4.0', sha256='eafb27f1ffbed427922ebe2b5b95d1c9c09bfeb897518867444fe230e3e35e41')
+    version('3.3.0', sha256='070530a472e7e78492f1f142c8d4b77c64de4626c4973b0589f0d18e1fcf5b4f')
+    version('3.2.1', sha256='d4388fb5f7e0083f0c344354c9ad3b5b823e2f3f27980e56efa7785140c9b616')
+    version('3.1.1', sha256='7f0df46458d8894ba0c2071b0848895304ae6b1137d3d4630f1600ed8eddf1a4')
```

Comment:
I think these versions should be removed from the base class and left only in the `singularity` package. If I try to get them from Sylabs fork I get:
```
$ wget https://github.com/sylabs/singularity/releases/download/v3.6.4/singularity-3.6.4.tar.gz
--2021-06-02 10:09:03--  https://github.com/sylabs/singularity/releases/download/v3.6.4/singularity-3.6.4.tar.gz
Resolving github.com (github.com)... 140.82.121.4
Connecting to github.com (github.com)|140.82.121.4|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2021-06-02 10:09:03 ERROR 404: Not Found.
```
which means, I think, the history remained in the `hpcng` repository.

Reply:
oh interesting - it makes total sense that releases didn't go with the fork / clone and push but I didn't verify it. Yes agreed we should remove these from singularityce. And ping @dtrudg in case there is a way to add back releases in the future.

Reply:
Huh this doesn't make sense, the releases are all there: https://github.com/sylabs/singularity/releases

Reply:
@alalazo what if we do:

```bash
$ wget https://github.com/sylabs/singularity/archive/refs/tags/v3.6.4.tar.gz
--2021-06-02 09:58:55--  https://github.com/sylabs/singularity/archive/refs/tags/v3.6.4.tar.gz
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/sylabs/singularity/tar.gz/refs/tags/v3.6.4 [following]
--2021-06-02 09:58:56--  https://codeload.github.com/sylabs/singularity/tar.gz/refs/tags/v3.6.4
Resolving codeload.github.com (codeload.github.com)... 140.82.114.9
Connecting to codeload.github.com (codeload.github.com)|140.82.114.9|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘v3.6.4.tar.gz’

v3.6.4.tar.gz                                  [   <=>                                                                                   ]   1.57M  2.89MB/s    in 0.5s    

2021-06-02 09:58:57 (2.89 MB/s) - ‘v3.6.4.tar.gz’ saved [1643796]
```
for the URL instead?

Reply:
Hmm but the sha256sum don't match:

```bash
$ sha256sum v3.6.4.tar.gz 
5995799213ebec2b2276cc1d238f3d5f0db985d42423e6dff4fb1664a91b2fe2  v3.6.4.tar.gz
```
I find that really strange... why would the hashes change?

Reply:
oh I see, it's the source code (and the releases are gone). That kind of sucks. :( I'll add 3.8.0 which does seem to be there!

Reply:
@vsoch I think what you'll find in the Sylabs fork are tarballs autogenerated by Github vs. tarballs put there by maintainers manually during the release process for hpcng

Reply:
Ahah, I love that due to browser caches you replied my comment in the past (i.e. you had it figured out already) :laughing:

Reply:
> oh interesting - it makes total sense that releases didn't go with the fork / clone and push but I didn't verify it. Yes agreed we should remove these from singularityce. And ping @dtrudg in case there is a way to add back releases in the future.

Hi @vsoch @alalazo - this was a bit of a difficult area with the releases. All the releases moved from sylabs/singularity -> hpcng/singularity a year ago. When we forked we just get the repo, not the issues, release etc.

We had two concerns:

1. We knew some people were still pulling from sylabs/singularity having not updated to use hpcng/singularity URLs.
2. We didn't want to mislead people about the origin of the older releases. Ours repo *is* now a fork, which is effective at the 3.8.0 CE release.

What we chose to do in the end, was to post only the most recent stable release (3.7.3, and later 3.7.4) as an exact mirror of the hpcng/singularity files, with a clear link to the hpcng location. The thinking being that people arriving on old links to the repo / release page would be able to get the tar if that's all they cared about, or follow the link to hpcng if they wanted to do that.

In retrospect we should have looked more into the situation for package managers automatically downloading the tarballs, and given a heads up. The GitHub auto-generated tarballs from tags are also a bit of a pain... since those have never been official release artifacts, but they do exist for all tags.

I don't think we will be likely to add the older releases back unless there's broad agreement in the hpcng community also that this would be more valuable than misleading.

Sorry for the trouble this has caused.

Reply:
It's totally okay! We are good to keep the older releases with hpcng, and add updated ones to singularityce moving forward. Thank you for the detail!

Reply:
I think there was some rebase issue and old versions are still in the `SingularityBase` class :grimacing: